e2e/src/components/PostsPreview.ts

```diff
@@ -74,4 +74,51 @@ export default class PostsPreview extends BaseComponent {
         const readTime = await getPreviewReadTime.innerText();
         return readTime;
     }
+
+    get allPosts(): Locator {
+        return this.postLocator;
+    }
+
+    get allReadMoreLinks(): Locator {
+        return this.page.locator('.read-more');
+    }
+
+    public async getAllPosts(): Promise<Locator[]> {
+        return await this.postLocator.all();
+    }
+
+    public async getAllReadMoreLinks(): Promise<Locator[]> {
+        return this.readMoreLinkLocator.all();
+    }
+
+    public async elementsAreVisible(elements: Locator): Promise<void> {
+        for (let i = 0; i < (await elements.count()); i++) {
+            const element = elements.nth(i);
+            await expect(element).toBeVisible();
+        }
+    }
+
+    async getAllLinks(allReadMoreLinks: Locator): Promise<Set<string>> {
+        const links = allReadMoreLinks.getByRole('link');
+        const allLinks = await links.all();
+        const allhrefs = await Promise.all(
+            allLinks.map((link) => link.getAttribute('href'))
+        );
+        const allValidHrefs = allhrefs.reduce((links, link) => {
+            expect.soft(link).toBeTruthy();
+            if (link) {
+                links.add(new URL(link, this.page.url()).href);
+            }
+            return links;
+        }, new Set<string>());
+        return allValidHrefs;
+    }
+
+    public async verifyLinksResponse(linksUrls: Set<string>): Promise<void> {
+        for(const url of linksUrls) {
+            const response = await this.page.request.get(url);
+            const isSuccessful = (response.status() === 200 || response.status() === 999);
+            expect.soft(isSuccessful).toBeTruthy();
+        }
+    }
```

Comment:
we have the same logic here >>> 
`e2e/src/components/Footer.ts`

<img width="761" alt="Screenshot 2025-02-12 at 08 59 43" src="https://github.com/user-attachments/assets/8da1d170-f04d-4549-b428-4f3b4dfb5328" />